ksqldb-functional-tests/src/test/resources/query-validation-tests/group-by.json

```diff
@@ -2272,6 +2313,277 @@
         {"topic": "OUTPUT", "key": false, "value": {"COUNT": 1}},
         {"topic": "OUTPUT", "key": true, "value": {"COUNT": 2}}
       ]
+    },
+    {
+      "properties": {
+        "ksql.key.format.enabled": "true"
+      },
+      "name": "AVRO primitive key",
+      "statements": [
+        "CREATE STREAM TEST (ID INT KEY, VAL BOOLEAN) WITH (kafka_topic='test_topic', format='AVRO');",
+        "CREATE TABLE OUTPUT AS SELECT VAL, COUNT() AS COUNT FROM TEST GROUP BY VAL;"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "keySchema": {"type": "int"},
+          "keyFormat": "AVRO",
+          "valueSchema": {"name": "ignored", "type": "record", "fields": [{"name": "VAL", "type": "boolean"}]},
+          "valueFormat": "AVRO"
+        }
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": {"VAL": true}},
+        {"topic": "test_topic", "key": 1, "value": {"VAL": false}},
+        {"topic": "test_topic", "key": 2, "value": {"VAL": true}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": true, "value": {"COUNT": 1}},
+        {"topic": "OUTPUT", "key": false, "value": {"COUNT": 1}},
+        {"topic": "OUTPUT", "key": true, "value": {"COUNT": 2}}
+      ],
+      "post": {
+        "sources": [
+          {
+            "comment": "unwrap singles is implicitly set on AVRO key formats",
```

Comment:
nit: unwrap singles is implicitly set on all key formats that support both wrapping and unwrapping, including AVRO. (I was confused for a moment when I read first this since I thought Avro was somehow special.)